railgun/internal/cmd/rootcmd/servers.go

```diff
@@ -45,23 +45,32 @@ func StartAdmissionServer(ctx context.Context, c *manager.Config) error {
 	return nil
 }
 
-func StartProfilingServer(ctx context.Context, c *manager.Config) error {
+func StartDiagnosticsServer(ctx context.Context, port int, c *manager.Config) (diagnostics.Server, error) {
 	deprecatedLogger, err := util.MakeLogger(c.LogLevel, c.LogFormat)
 	if err != nil {
-		return err
+		return diagnostics.Server{}, err
 	}
 	logger := logrusr.NewLogger(deprecatedLogger)
 
-	if !c.EnableProfiling {
-		logger.Info("profiling server disabled")
-		return nil
+	if !c.EnableProfiling && !c.EnableConfigDumps {
+		logger.Info("diagnostics server disabled")
+		return diagnostics.Server{}, nil
 	}
 
-	s := diagnostics.Server{Logger: logger}
+	s := diagnostics.Server{
+		Logger:           logger,
+		ProfilingEnabled: c.EnableProfiling,
+	}
+	if c.EnableConfigDumps {
+		s.ConfigDumps = util.ConfigDumpDiagnostic{
+			DumpsIncludeSensitive: c.DumpSensitiveConfig,
+			Configs:               make(chan util.ConfigDump, 3),
```

Comment:
How strongly are we feeling that 3 is a good buffer number?

At minimum, I would like to see this number stored in a var for the package with associated documentation that will enable the reader to better understand the reasoning why this was chosen, and the effects of modifying it so that it doesn't become esoteric.

Ultimately though, if we have some sense of the upper bounds of memory usage for these config dumps we should make our default based on a total memory storage tolerance for the queue and document that in the var.

Reply:
Not much. 28a2914e to document it more clearly, but the choice is basically arbitrary. We'd only know if it's too small a choice if we get field reports indicating that diagnostic configs were getting dropped.

Memory usage depends on config size and has no known upper bound, but this isn't a feature we intend for normal usage, so I'm not too concerned about that. A temporary increase in usage during a test run is fine.

Reply:
Having documented so the next reader can better understand, I think this is OK for this iteration :+1: